pyproject.toml

```diff
@@ -0,0 +1,97 @@
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# SETUP CONFIGURATION.                                                        #
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# ISORT CONFIGURATION.                                                        #
+[tool.isort]
+profile = "black"
+known_first_party = ["ote_cli", "ote_sdk"]
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# MYPY CONFIGURATION.                                                         #
+[tool.mypy]
+python_version = 3.8
+ignore_missing_imports = true
+show_error_codes = true
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# PYLINT CONFIGURATION                                                        #
+[tool.pylint.master]
+ignore = "CVS"
+ignore-patterns = "^.*test_.*.py"
+
+[tool.pylint.messages_control]
+disable = [
+    "logging-fstring-interpolation",
+    "too-few-public-methods",
+    "fixme",
+    "duplicate-code",
+    "raw-checker-failed",
+    "bad-inline-option",
+    "locally-disabled",
+    "file-ignored",
+    "suppressed-message",
+    "useless-suppression",
+    "deprecated-pragma",
+    "use-symbolic-message-instead",
+    "import-outside-toplevel",
+    "arguments-differ",
+]
+
+[tool.pylint.typecheck]
+generated-members = ["numpy.*", "torch.*"]
+ignored-modules = [
+    "ote_sdk",
```

Comment:
> I feel we should have setup.py first as currently tox requires a base env with everything installed. But that might be a lot more effort.
This draft PR adds https://github.com/openvinotoolkit/training_extensions/pull/1199 `setup.py`

Reply:
> I would be in favour of treating ote_sdk as first party and fixing the isort issues first as we might forget about these settings in the future. Plus it would make it easier to move cli and sdk into a single package.

I also thought about that, but gave moving `ote.cli` a try, which seemed to sort imports ok so far.